src/ast/query.rs

```diff
@@ -298,14 +298,7 @@ impl fmt::Display for Select {
         if let Some(ref selection) = self.selection {
             write!(f, " WHERE {selection}")?;
         }
-        match &self.group_by {
-            GroupByExpr::All => write!(f, " GROUP BY ALL")?,
-            GroupByExpr::Expressions(exprs) => {
-                if !exprs.is_empty() {
-                    write!(f, " GROUP BY {}", display_comma_separated(exprs))?;
-                }
-            }
-        }
+        write!(f, "{}", self.group_by)?;
```

Comment:
```suggestion
        write!(f, " {}", self.group_by)?;
```
Can we change the group_by display method to not include a leading space? its easier to have the caller insert the leading space, that way the display method is agnostic of where its being placed (and can be reused in different contexts as a result), and in this case each individual variant wont have to include its own leading space